core/src/main/java/com/amplitude/core/Storage.kt

```diff
@@ -18,6 +19,8 @@ interface Storage {
 
     suspend fun writeEvent(event: BaseEvent)
 
+    suspend fun writeEvent(event: JSONObject)
```

Comment:
@falconandy I'm guessing this was added to prevent double conversion of events during migration (JSON -> BaseEvent - JSON) but am a little concerned about adding JSONObject to the core Storage interface.

For example, JRE doesn't include JSON in the core libraries and it would need to be added as an additional dependency. The goal for this repo in the long run is to support both Android and JRE and this could complicate things.

Since the migration plugin (hopefully) only runs once, I think I would prefer the double conversion over adjusting the core Storage interface.

wdyt?

cc @qingzhuozhen @liuyang1520

Reply:
@justin-fiedler Currently `core` package already depends on `org.json.*` types (JSONException, JSONObject, JSONArray).
Adding JSONObject to the core Storage interface doesn't change anything - JRE should add JSON as an additional dependency in any case.
wdyt?

Reply:
Hmm, got it. In that cause it matters less, but it still seems "less clean" to have to support JSON in addition and BaseEvent to define a storage.

I still would prefer double conversion over adding JSON to Storage interface. Let me know if you strongly disagree, thanks.

Reply:
Possible issue is differences in json encoding between legacy/current SDK. I added code to fix some ones (`timestamp` -> `time`, `uuid` -> `insert_id`, `library`) but maybe `legacy JSON` ->`event` -> `current JSON` looks/works better

Reply:
Thanks, I saw the conversion code. I think you are handling the most important fields.

> legacy JSON ->event -> current JSON

This sounds good 👍 if we find any non-transferred fields in the future we can update the conversion code as needed.